projects/angular-cesium/src/lib/angular-cesium-widgets/components/toolbar/ac-toolbar/ac-toolbar.component.ts

```diff
@@ -47,41 +58,67 @@ import { CesiumService } from '../../../../angular-cesium/services/cesium/cesium
     changeDetection: ChangeDetectionStrategy.OnPush,
   }
 )
-export class AcToolbarComponent implements OnInit, OnDestroy {
+export class AcToolbarComponent implements OnInit, OnChanges, OnDestroy {
   @Input()
   toolbarClass: string;
   @Input()
   allowDrag = true;
+  @Output()
+  onDrag = new EventEmitter<MouseEvent>();
 
   dragStyle = {
     'height.px': 20,
     'width.px': 20,
   };
 
-  private subscription: Subscription;
+  private mouseDown$: Observable<MouseEvent>;
+  private mouseMove$: Observable<MouseEvent>;
+  private mouseUp$: Observable<MouseEvent>;
+  private drag$: Observable<MouseEvent>;
+  private dragSubscription: Subscription;
 
-  constructor(private element: ElementRef, private cesiumService: CesiumService) {
-  }
+  constructor(private element: ElementRef, private cesiumService: CesiumService) {}
 
   ngOnInit() {
     this.cesiumService.getMap().getMapContainer().appendChild(this.element.nativeElement);
     if (this.allowDrag) {
-      const mouseDown$ = observableFromEvent(this.element.nativeElement, 'mousedown');
-      const mouseMove$ = observableFromEvent(document, 'mousemove');
-      const mouseUp$ = observableFromEvent(document, 'mouseup');
+      this.listenForDragging();
+    }
+  }
 
-      const drag$ = mouseDown$.pipe(switchMap(() => mouseMove$.pipe(takeUntil(mouseUp$))));
 
-      this.subscription = drag$.subscribe((event: MouseEvent) => {
-        this.element.nativeElement.style.left = event.x + 'px';
-        this.element.nativeElement.style.top = event.y + 'px';
-      });
+  ngOnChanges(changes: SimpleChanges) {
```

Comment:
Simple boolean logic here to indicate whether or not the `allowDrag` input is changing - we listen or unsubscribe accordingly.